services/tables/src/main/java/com/linkedin/openhouse/tables/api/validator/impl/PoliciesSpecValidator.java

```diff
@@ -86,9 +87,71 @@ public boolean validate(
         return false;
       }
     }
+
+    return validateReplication(policies, tableUri);
+  }
+
+  /**
+   * Valid cases for replication object: 0. Interval input can be either be accepted as 12H or daily
+   * from 1-3D 1. Destination cluster cannot be equal to the source cluster
+   */
+  protected boolean validateReplication(Policies policies, TableUri tableUri) {
+    if (policies != null
+        && policies.getReplication() != null
+        && policies.getReplication().getConfig() != null) {
+      return policies.getReplication().getConfig().stream()
+          .allMatch(
+              replicationConfig -> {
+                if (replicationConfig.getInterval() != null
+                    && !replicationConfig.getInterval().isEmpty()) {
+                  if (!validateReplicationInterval(replicationConfig)) {
+                    failureMessage =
+                        String.format(
+                            "Replication interval for the table [%s] can either be 12 hours or daily for up to 3 days",
+                            tableUri);
+                    return false;
+                  }
+                }
+                if (replicationConfig.getDestination() != null) {
+                  if (!validateReplicationDestination(replicationConfig, tableUri)) {
+                    failureMessage =
+                        String.format(
+                            "Replication destination cluster for the table [%s] must be different from the source cluster",
+                            tableUri);
+                    return false;
+                  }
+                }
+                return true;
+              });
+    }
     return true;
   }
 
+  /**
+   * Validate that the optional interval parameter provided by users exists as an interval of 12 or
+   * as a daily value up to 3 days
+   */
+  protected boolean validateReplicationInterval(ReplicationConfig replicationConfig) {
+    String granularity =
+        replicationConfig.getInterval().substring(replicationConfig.getInterval().length() - 1);
+    int interval =
+        Integer.parseInt(
+            replicationConfig
+                .getInterval()
+                .substring(0, replicationConfig.getInterval().length() - 1));
+
+    return (interval >= 0 && interval <= 3 && granularity.equals("D"))
```

Comment:
This condition will accept `0D` as input as well right? How are we interpreting that?

Reply:
yeah, thanks for catching that. updated